javascript/javascript2/week2/movies.js

```diff
@@ -0,0 +1,107 @@
+
+
+//2-Create an array of movie titles with long movie titles. long title means if title length is more than 15
+
+const LongMovieTitle = movies.filter(function(movie){
+    return movie.title.length > 15;
+}).map((movie) => movie.title);
+console.log("Long Movies Title: ", LongMovieTitle);
+
+//3-Count the number of movies made between 1980-1989 (including both the years)
+
+const numOfMovie = movies.filter((movie) => movie.year > 1980 && movie.year <1989);
+const countOfMovie = numOfMovie.length;
+console.log("Count of movies: ", countOfMovie); 
+
+//4-Create a new array that has an extra key called tag. The tag is based on the rating: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)
+
+const moviesByRating = movies.filter(function(movie){
+    if (movie.rating >= 7)
+    {
+        return movie.tag = "Good";
+    }
+    if( movie.rating >= 4 && movie.rating < 7)
+    {
+        return movie.tag = "Average";
+    }
+    if(movie.rating <4)
+    {
+        return movie.tag = "Bad";
+    }
+});
+console.log("movies by rating",moviesByRating);
+
+// 5 - Using chaining, first filter the movies array to only contain the movies rated higher than 6.
+
+const movieHigerRating = movies.filter(movie => movie.rating > 6).map(movie => movie.rating);
+console.log(" movies rated higher than 6", movieHigerRating)
+
+//6-Count the total number of movies containing any of following keywords: Surfer, Alien or Benjamin.
+
+const findName = movies.filter(function(movie){
+    
+    if((movie.title.toLowerCase().indexOf("surfer") >= 0) || 
+    (movie.title.toLowerCase().indexOf("alien") >= 0) ||
+    (movie.title.toLowerCase().indexOf("benjamin") >= 0))
+    {
+        return true;
+    }    
+});
+console.log("Total Number of movie that contain Surfer, Alien, Benjamin",findName.length)
+
+//duplicated words in the title: 
+
+const movieTitile = movies.map(movie => movie.title)
+.filter((title) => findDuplicateWord(title));
+console.log(movieTitile);
+
+function findDuplicateWord(stringName)
+{
+    let found = false;
+    const nameArr = stringName.split(" ");
+    const len = nameArr.length;
+    for(i =0; i <len ; i++)
+    {
+       const removedItem = nameArr.splice(0, 1);//nameArr[0];
+        if(nameArr.includes(removedItem[0]))
+        {
+            found = true;
+            break;
+        }           
+    } 
+    return found;
+}
+
+//Calculate the average rating of all the movies using reduce. Optional
+
+const averageRatingOfAllMovie = movies.map(movie => movie.rating);
+const moviescount = averageRatingOfAllMovie.length;
+function sum(averageRatingOfAllMovie) {
+    const reducer = (accumulator, currentValue) => accumulator + currentValue;
+    const initialValue = 0;
+    return averageRatingOfAllMovie.reduce(reducer, initialValue);
+  }
+const movieRatingSum = sum(averageRatingOfAllMovie)
+const moviesAverageRating = movieRatingSum / moviescount;
+console.log("All Movies Average Rating :", moviesAverageRating.toFixed(1));
```

Comment:
```js
const summedRatings = movies.reduce((sum, movie) =>  return sum + movie, 0);
const averageRating = summedRatings / movies.length
```

Reply:
yes, I fixed it too. if everything looks ok. may I merge in master.

Reply:
If you're happy with you work :) You don't need my permission